lib/talkable/generators/install_generator.rb

```diff
@@ -37,20 +37,45 @@ def load_talkable_offer
 
       empty_directory "app/views/shared/"
 
-      if options[:haml] || options[:slim]
-        ext = options[:haml] ? 'haml' : 'slim'
+      if erb?
+        copy_file "app/views/shared/_talkable_offer.html.erb", "app/views/shared/_talkable_offer.html.erb"
+        inject_into_file "app/views/layouts/application.html.erb", before: "</body>" do
+          "<%= render 'shared/talkable_offer', offer: @offer %>\n"
+        end
+      else
+        ext = template_lang
 
         copy_file "app/views/shared/_talkable_offer.html.#{ext}", "app/views/shared/_talkable_offer.html.#{ext}"
         gsub_file "app/views/layouts/application.html.#{ext}", /^(\s*)\=\s*yield\s*$/ do |line|
           paddings = line.match(/(\s*)\=/)[1]
-          "#{line}#{paddings}= render 'shared/talkable_offer'\n"
+          "#{line}#{paddings}= render 'shared/talkable_offer', offer: @offer\n"
         end
+      end
+    end
+
+    def add_invite_controller
+      ext = template_lang
+
+      copy_file "app/controllers/invite_controller.rb", "app/controllers/invite_controller.rb"
+      copy_file "app/views/invite/show_offer.html.#{ext}", "app/views/invite/show_offer.html.#{ext}"
+      route "get '/invite' => 'invite#show_offer'"
+    end
+
+    protected
+
+    def template_lang
+      @template_lang ||= if options[:haml]
+        'haml'
+      elsif options[:slim]
+        'slim'
```

Comment:
I don't believe these stats. They only show the github activity. I think
the real use for haml over slim is at least 5x.

2016-09-01 14:58 GMT+03:00 Alex Tonkonozhenko <notifications@github.com>:

> In lib/talkable/generators/install_generator.rb
> <https://github.com/talkable/talkable-ruby/pull/9#discussion_r77159885>:
>
> > +
> > +    def add_invite_controller
> > +      ext = template_lang
> > +
> > +      copy_file "app/controllers/invite_controller.rb", "app/controllers/invite_controller.rb"
> > +      copy_file "app/views/invite/show_offer.html.#{ext}", "app/views/invite/show_offer.html.#{ext}"
> > +      route "get '/invite' => 'invite#show_offer'"
> > +    end
> > +
> > +    protected
> > +
> > +    def template_lang
> > +      @template_lang ||= if options[:haml]
> > +        'haml'
> > +      elsif options[:slim]
> > +        'slim'
>
> It's not rare. The difference is not large. https://www.ruby-toolbox.com/
> categories/template_engines
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/talkable/talkable-ruby/pull/9/files/4bfc12aa6c932d1f23815663c5817dded7a0c5ff#r77159885>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHeRLKDVB0K4oFU0oNp5Z-wQseWQUYsks5qlr3TgaJpZM4Jymbs>
> .
>



-- 
Bogdan Gusiev.
agresso@gmail.com